### PR TITLE
code-of-conduct file updated

### LINKED ISSUE #1: Add Details To Code of Conduct File
The code_of_conduct.md file is currently empty. I propose adding in the details to outline the expected behaviour of contributors clearly. 

**Expected Outcome:**

- Add the details of the code_of_conduct .md file to the repository's root directory. 
- Ensure that the guidelines are clear, inclusive, and representative of the project's values.

**Request**: 
Please assign this issue  to me, as I would like to contribute by filling up this file. 

**Labels**: 
Documentation, Enchancement

### PR DESCRIPTION
**Pull request by tosan-okome**
**Title**: 

Add Code of Conduct for Project

**Description**:
This pull request updates the `code_of_conduct.md` file to establish a safe and respectful environment for all project contributors. 

**Related Issues**:
> Closes #1 

**Additional Notes**:
Feel free to provide feedback or suggest additional points to be included in the Code of Conduct. This document is intended to evolve with the community’s needs.

### CODE REVIEW

File: code-of-conduct.md [hunk 1]
```diff
@@ -1 +1,54 @@
...
+
+### Our Standards
+To promote a positive and productive environment, participants are expected to:
+- Use welcoming and respectful language.
+- Be considerate of differing viewpoints, experiences, and backgrounds.
+- Gracefully accept constructive criticism and strive to learn from mistakes.
+- Demonstrate empathy and kindness toward others.
+- Focus on the collective success of the community and project.
+
+Unacceptable behaviours include but are not limited to:
```
Comment (NPDebs) [inline code-of-conduct.md:14]:
```suggestion
**Unacceptable behaviours include but are not limited to:**
```

File: code-of-conduct.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
+- The use of sexualized language, images, or unwelcome sexual attention or advances.
+- Insults, derogatory comments, trolling, or personal and political attacks.
+- Harassment, intimidation, stalking, or inappropriate social contact.
+- Publishing others’ private information, such as a physical or electronic address, without explicit permission.
+- Threats of violence, incitement to violence, or encouraging self-harm.
+- Discriminatory jokes or using derogatory slurs related to identity.
+- Sustained disruption of discussions, either in public or private channels.
+
+### Scope
+This Code of Conduct applies to all spaces associated with the project, including but not limited to project repositories, mailing lists, issue trackers, events (online or offline), and public representation of the project in other venues, such as social media. This also includes interactions outside of project spaces when they affect the community, such as harassment at events or other forms of misconduct that may harm our community's reputation.
```
Comment (NPDebs) [inline code-of-conduct.md:24]:
```suggestion
This Code of Conduct applies to **all spaces** associated with the project, including but not limited to project repositories, mailing lists, issue trackers, events (online or offline), and public representation of the project in other venues, such as social media. This also includes interactions outside of project spaces when they affect the community, such as harassment at events or other forms of misconduct that may harm our community's reputation.
```

File: code-of-conduct.md [hunk 3]
```diff
@@ -1 +1,54 @@
...
+
+Certain project spaces may have additional rules specific to their nature, which participants are expected to follow alongside this Code of Conduct.
+
+### Enforcement
+Project maintainers are responsible for enforcing this Code of Conduct and ensuring compliance in project spaces. They are empowered to take corrective actions, which may include:
```
Comment (NPDebs) [inline code-of-conduct.md:29]:
```suggestion
**Project maintainers** are responsible for enforcing this Code of Conduct and ensuring compliance in project spaces. They are empowered to take corrective actions, which may include:
```

File: code-of-conduct.md [hunk 4]
```diff
@@ -1 +1,54 @@
...
+
+- debs@we-are-ols.org
+- okomebtosan@gmail.com
+
+When reporting, please provide as much information as possible, including:
+- Your contact details.
+- A description of the incident, including any individuals involved, witnesses, and if the incident is ongoing.
+- Any supporting documentation, such as screenshots or relevant links.
+
+All complaints will be handled with discretion and confidentiality. If the report involves a member of the response team, they will recuse themselves from handling the issue. In such cases, another member of the project team will manage the report.
```
Comment (NPDebs) [inline code-of-conduct.md:45]:
```suggestion
**All complaints will be handled with discretion and confidentiality**. If the report involves a member of the response team, they will recuse themselves from handling the issue. In such cases, another member of the project team will manage the report.
```

File: code-of-conduct.md [hunk 5]
```diff
@@ -1 +1,54 @@
...
+
+### Confidentiality and Transparency
+We are committed to maintaining the confidentiality of reporters and victims. In certain situations, if it’s necessary to ensure the safety of the community or the general public, we may take further action by privately warning third parties or publicly naming the individual responsible for unacceptable behaviour. We will only name victims with their affirmative consent.
+
+### Acknowledgement of Mistakes
+Our community seeks to treat everyone as fairly and equally as possible. When mistakes happen, we expect members to take responsibility for their actions. If someone is harmed or offended, we are committed to listening carefully and respectfully, learning from the incident, and taking steps to make things right.
+
+### Attribution
+This Code of Conduct is adapted from various open-source community codes, including the Open Life Science, Contributor Covenant (version 2.0), Geek Feminism, and CHAOSS, which promote inclusive and harassment-free participation.
```
Comment (NPDebs) [inline code-of-conduct.md:54]:
```suggestion
This Code of Conduct is adapted from various open-source community codes, including the [Open Life Science](https://github.com/open-life-science/open-life-science.github.io), [Contributor Covenant](https://www.contributor-covenant.org/)(version 2.0), [Geek Feminism](https://geekfeminismdotorg.wordpress.com/about/code-of-conduct/), and [CHAOSS](https://chaoss.community/), which promote inclusive and harassment-free participation.
```

**Review by NPDebs — APPROVED** [review]
@TeeOkome - First of all, congrats on snagging the first issue and PR in this repo!! 🎉👏  

This is robust, well-rounded and beatifully captures the values that shape a thriving community! ❤️ Thank you for putting so much thought into it!

I left a few suggestions to mainly:
- format the text to emphasise some important words,
- link to the communities you credited.

I'm more than happy to merge once you reject or accept the suggestions. Thanks again!

**Comment by TeeOkome** [comment]
@NPDebs 
These are very handy suggestions. All accepted.  
Thanks for the extra eye!

Also, "#1" does have a nice ring to it. 😄

**Comment by NPDebs** [comment]
Thanks for your contribution, @TeeOkome! ✨🚀

**Pull request merged**